src/main/scala/com/mini/shop/Application.scala

```diff
@@ -1,8 +1,19 @@
 package com.mini.shop
 
-object Application{
+import com.mini.shop.repo.{ItemRepo, OfferRepo}
+import com.mini.shop.service.CheckoutService
 
+object Application
+{
   def main(args: Array[String]): Unit = {
-    println("Hello World!")
+    /*TBD: Object initialization can later be used via dependency injections*/
+    val itemRepo = new ItemRepo
+    val offerRepo = new OfferRepo
+    val checkoutService = new CheckoutService(itemRepo, offerRepo)
+    val inputItems = List("Apple", "Apple", "Orange", "Apple")
```

Comment:
for now i guess creating these repo and service classes is an overkill, i believe 
for interview purpose we should always keep it simple